extensions/cli/cli.go

```diff
@@ -0,0 +1,109 @@
+package cli
+
+import (
+	ranchercli "github.com/rancher/shepherd/clients/ranchercli"
+	"github.com/rancher/shepherd/pkg/session"
+	"github.com/sirupsen/logrus"
+)
+
+// SwitchContext will display the current context and switch to the default one.
+func SwitchContext(session *session.Session) error {
+	client := &ranchercli.Client{
+		Session: session,
+	}
+
+	logrus.Infof("Listing the current context...")
+	err := client.Create("./rancher", "context", "current")
+	if err != nil {
+		return err
+	}
+
+	logrus.Infof("Switching to the default context...")
+	err = client.Create("./rancher", "context", "switch")
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// CreateAndDeleteProjects will create and delete projects in the local cluster.
+func CreateAndDeleteProjects(session *session.Session) error {
+	client := &ranchercli.Client{
+		Session: session,
+	}
+
+	logrus.Infof("Creating a project...")
+	err := client.Create("./rancher", "projects", "create", "--cluster", "local", "auto-cli-project")
+	if err != nil {
+		return err
+	}
+
+	logrus.Infof("Deleting the project...")
+	err = client.Create("./rancher", "projects", "delete", "auto-cli-project")
+	if err != nil {
```

Comment:
The idea behind the Rancher CLI is that it's an alternative method to do calls outside of Norman, Steve, etc. So users that use Rancher CLI wouldn't mix and match. So for rancher projects, it would just be the command `rancher projects ls` to see if what we just made works.

From the points above, that isn't going to give what you want, but that is how a CLI user would validate their projects are existing.

Reply:
I am okay with using the CLI to list what is created, as long as there's some validation after the creation call I'm good -- I do think steve would be the proper way to QA it but I can see your point too so whatever you're comfortable adding